src/dotnet/Telemetry.cs

```diff
@@ -146,5 +155,49 @@ private void TrackEventTask(string eventName, IDictionary<string, string> proper
                 return _commonProperties;
             }
         }
+        
+        private List<string> GetMacs()
+        {
+            NetworkInterface[] nics = NetworkInterface.GetAllNetworkInterfaces();
+            var macs = new List<string>();
+            
+            if (nics == null || nics.Length < 1)
+            {
+                macs.Add(string.Empty);
+                return macs;
+            }
+            
+            foreach (NetworkInterface adapter in nics)
+            {
+                IPInterfaceProperties properties = adapter.GetIPProperties();
+
+                PhysicalAddress address = adapter.GetPhysicalAddress();
+                byte[] bytes = address.GetAddressBytes();
+                macs.Add(string.Join("-", bytes.Select(x => x.ToString("X2"))));
```

Comment:
The reason they are getting turned into a string and back again, is an attempt to match and hash a specific format (a string in form of FF-FF-FF-FF-FF-FF) in as most similar way that VsCode hashes their mac address. 

Essentially I am trying to duplicate the following
[VsCode hashing](https://github.com/Microsoft/vscode/blob/224bbb8b453715395ec5e7b7a6dc546f7eb2c08b/src/vs/platform/telemetry/node/commonProperties.ts)
e.g. 
result = crypto.createHash('sha256').update(macAddress, 'utf8').digest('hex');
Which gets the 
[macAddress ](https://github.com/bevry/getmac/blob/master/src/lib/getmac.coffee) from a regex match from the string of getmac or ifconfig


If there is a way I can do this in a better way without the back and forth. I'd love to improve it, I am just not sure how to do this any other way.